cpp/include/cudf/detail/aggregation/aggregation.hpp

```diff
@@ -27,6 +27,63 @@
 
 namespace cudf {
 namespace detail {
+
+class aggregation_finalizer;
+
+class compound_aggregation {
+ public:
+  virtual std::vector<aggregation> get_simple_aggregations(data_type col_type) const { return {}; }
+  virtual void finalize(aggregation_finalizer& finalizer) = 0;
+};
+
+// Forward declare compound aggregations.
+class mean_aggregation;
+class std_var_aggregation;
+class min_aggregation;
+class max_aggregation;
+
+// Visitor pattern
+class aggregation_finalizer {  // Declares the interface for the finalizer
+ public:
+  // Declare overloads for each kind of a agg to dispatch
+  virtual void Dispatch(min_aggregation& agg)     = 0;
```

Comment:
Nice use of visitor pattern. I'd maybe rename `Dispatch` to `visit` to be more canonical with the visitor pattern.

Reply:
Done